### PR TITLE
Introduce eslint import rules

### PR DESCRIPTION
**Pull request by matikrk**

### CODE REVIEW

File: eslint/main.js [hunk 1]
```diff
@@ -524,6 +527,50 @@ module.exports = {
...
-    'yield-star-spacing': ['error', 'after']
+    'yield-star-spacing': ['error', 'after'],
+
+    // IMPORTS
+    // This rule reports any imports that come after non-import statements.
+    'import/first': 'error',
+    // Reports if a resolved path is imported more than once.
+    'import/no-duplicates': 'error',
+
+    // TODO  Not working now :/ check in next version, coz very nice rule
```
Comment (vergilius) [inline eslint/main.js]: remove me

Comment (vergilius) [reply eslint/main.js]: I don't think it's nice rule :)

File: eslint/main.js [hunk 2]
```diff
@@ -524,6 +527,50 @@ module.exports = {
...
+    // issue on github: https://github.com/benmosher/eslint-plugin-import/issues/953
+    // // This rule enforces that all exports are declared at the bottom of the file.
+    // 'import/exports-last': 'error',
+
+    // Ensure consistent use of file extension within the import path
+    'import/extensions': ['error', {
+      js: 'never',
```
Comment (vergilius) [inline eslint/main.js]: 👍

File: eslint/main.js [hunk 3]
```diff
@@ -524,6 +527,50 @@ module.exports = {
...
+      jsx: 'never'
+    }],
+    // Enforce a convention in module import order
+    'import/order': 'error',
+    // When there is only a single export from a module, prefer using default export over named export.
```
Comment (vergilius) [inline eslint/main.js]: I don't think it's good to do it this way, imagine utils file that you have single function to export, but in future there will be more of it

File: eslint/main.js [hunk 4]
```diff
@@ -524,6 +527,50 @@ module.exports = {
...
+    'import/prefer-default-export': 'error',
+    // Enforces having one or more empty lines after the last top-level import statement or require call.
+    'import/newline-after-import': 'error',
+    // Reports if a module's default export is unnamed
+    'import/no-anonymous-default-export': ['error', {
```
Comment (vergilius) [inline eslint/main.js]: hm, I don't like it, I'm exporting unnamed array in a few places (stuff like like subjects in actions)

File: eslint/main.js [hunk 5]
```diff
@@ -524,6 +527,50 @@ module.exports = {
...
+      'allowObject': false
+    }],
+    // Reports require([array], ...) and define([array], ...) function calls at the module scope.
+    'import/no-amd': 'error',
+    // Enforces names exist at the time they are dereferenced, when imported as a full namespace
+    'import/namespace': 'error',
+    // Reports funny business with exports, like repeated exports of names or defaults.
+    'import/export': 'error',
+    // Reports use of an exported name as the locally imported name of a default export.
+    'import/no-named-as-default': 'error',
```
Comment (vergilius) [inline eslint/main.js]: won't this be a problem withing style-guide?

File: eslint/main.js [hunk 6]
```diff
@@ -524,6 +527,44 @@ module.exports = {
...
+    'import/first': 'error',
+    // Reports if a resolved path is imported more than once.
+    'import/no-duplicates': 'error',
+    // Ensure consistent use of file extension within the import path
+    'import/extensions': ['error', {
+      js: 'never',
+      jsx: 'never'
+    }],
+    // Enforce a convention in module import order
+    'import/order': 'error',
```
Comment (aju) [inline eslint/main.js]: I think this is not the best option

File: eslint/main.js [hunk 7]
```diff
@@ -524,6 +527,44 @@ module.exports = {
...
+    // When there is only a single export from a module, prefer using default export over named export.
+    'import/prefer-default-export': 'error',
+    // Enforces having one or more empty lines after the last top-level import statement or require call.
+    'import/newline-after-import': 'error',
```
Comment (aju) [inline eslint/main.js]: 👍

File: eslint/main.js [hunk 8]
```diff
@@ -524,6 +527,44 @@ module.exports = {
...
+      'allowLiteral': false,
+      'allowObject': false
+    }],
+    // Reports require([array], ...) and define([array], ...) function calls at the module scope.
+    'import/no-amd': 'error',
+    // Enforces names exist at the time they are dereferenced, when imported as a full namespace
+    'import/namespace': 'error',
+    // Reports funny business with exports, like repeated exports of names or defaults.
+    'import/export': 'error',
+    // Reports use of an exported name as the locally imported name of a default export.
```
Comment (aju) [inline eslint/main.js]: hm?

File: eslint/main.js (hunk 6 above)
Comment (vergilius) [reply eslint/main.js]: I would love to see it somehow ordered like this:
- external imports (stuff from package.json),
- styleguide,
- local imports (config/utils)
- local imports (classes/components)

the truth is, first two and second two are rather not distinguishable :(

File: eslint/main.js (hunk 4 above)
Comment (matikrk) [reply eslint/main.js]: https://github.com/benmosher/eslint-plugin-import/blob/master/docs/rules/no-anonymous-default-export.md

most common:
`export default class {}`

do you write like this?
`export default []`
i think better is:
```const extraArray=[]
export default  extraArray```
especially when this array is long


We can specify that rule

File: package.json [hunk 9]
```diff
@@ -12,6 +12,7 @@
   "dependencies": {
     "eslint": "^4.5.0",
     "eslint-plugin-babel": "^4.1.2",
+    "eslint-plugin-import": "^2.7.0",
```
Comment (kdzwinel) [inline package.json]: no longer maintained as it was integrated into eslint - https://www.npmjs.com/package/eslint-plugin-import-order

Comment (matikrk) [reply package.json]: it's different plugin

File: eslint/main.js (hunk 4 above)
Comment (kdzwinel) [reply eslint/main.js]: Naming things can't do any harm, I think it's a good idea to keep it.

File: eslint/main.js (hunk 6 above)
Comment (matikrk) [reply eslint/main.js]: I also like ordered imports

File: eslint/main.js (hunk 8 above)
Comment (matikrk) [reply eslint/main.js]: It's  helpful  when you forget give brackets on named export

File: eslint/main.js (hunk 5 above)
Comment (matikrk) [reply eslint/main.js]: I will try everything defined now. If something will be too problematic I will change.

Some rules wasn't applicable to our projects

File: eslint/main.js (hunk 3 above)
Comment (matikrk) [reply eslint/main.js]: ok.

File: eslint/main.js (hunk 6 above)
Comment (vergilius) [reply eslint/main.js]: `The default value is ["builtin", "external", "parent", "sibling", "index"].`

Comment (aju) [reply eslint/main.js]: I think is alphabetical and that is not that good

Comment (vergilius) [reply eslint/main.js]: this might cause quite huge amount of changes, I'd suggest to disable it, and enable separately?

**Pull request merged**